Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCProducer.cc

```diff
@@ -0,0 +1,121 @@
+
+#include "TMath.h"
+//The class 
+class AlcaPCCProducer : public edm::one::EDProducer<edm::EndLuminosityBlockProducer,edm::one::WatchLuminosityBlocks>{
+  public:
+    explicit AlcaPCCProducer(const edm::ParameterSet&);
+    ~AlcaPCCProducer();
+
+  private:
+    virtual void beginLuminosityBlock     (edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup) ;
+    virtual void endLuminosityBlock       (edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup) ;
+    virtual void endLuminosityBlockProduce(edm::LuminosityBlock& lumiSeg, const edm::EventSetup& iSetup) ;
+    virtual void produce                  (edm::Event& iEvent, const edm::EventSetup& iSetup) ;
+ 
+    edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster> >  pixelToken;
+    edm::InputTag   fPixelClusterLabel;
+
+    std::string trigstring_;      //specifies the trigger Rand or ZeroBias 
+    int countEvt_;       //counter
+    int countLumi_;      //counter
+
+    std::unique_ptr<reco::PixelClusterCounts> thePCCob;
+
+};
+
+//--------------------------------------------------------------------------------------------------
+AlcaPCCProducer::AlcaPCCProducer(const edm::ParameterSet& iConfig)
+{
+    fPixelClusterLabel = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getParameter<edm::InputTag>("pixelClusterLabel");
+    trigstring_ = iConfig.getParameter<edm::ParameterSet>("AlcaPCCProducerParameters").getUntrackedParameter<std::string>("trigstring","alcaPCC");
+
+    countLumi_ = 0;
+
+    produces<reco::PixelClusterCounts, edm::InLumi>(trigstring_);
+    pixelToken=consumes<edmNew::DetSetVector<SiPixelCluster> >(fPixelClusterLabel);
+}
+
+//--------------------------------------------------------------------------------------------------
+AlcaPCCProducer::~AlcaPCCProducer(){
+}
+
+//--------------------------------------------------------------------------------------------------
+void AlcaPCCProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
+    countEvt_++;
+
+    unsigned int bx=iEvent.bunchCrossing();
+    //std::cout<<"The Bunch Crossing"<<bx<<std::endl;
+    thePCCob->eventCounter(bx);
+
+    //Looping over the clusters and adding the counts up  
+    edm::Handle< edmNew::DetSetVector<SiPixelCluster> > hClusterColl;
+    iEvent.getByToken(pixelToken,hClusterColl);
+
+    const edmNew::DetSetVector<SiPixelCluster>& clustColl = *(hClusterColl.product()); 
+    // ----------------------------------------------------------------------
+    // -- Clusters without tracks
+    for (auto const & mod: clustColl) {
+        if(mod.empty()) { continue; }
+        DetId detId = mod.id();
+
+        // -- clusters on this det
+        edmNew::DetSet<SiPixelCluster>::const_iterator  di;
+        int nClusterCount=0;
+        for (di = mod.begin(); di != mod.end(); ++di) {
+            nClusterCount++;
+        }
+        int nCluster = mod.size();
+        thePCCob->increment(detId(), bx, nCluster);
```

Comment:
ok, seems the answer is yes... perhaps the stanza about nClusterCount is obsolete?